## Soal Nomor 2
* Regresi menggunakan Algoritma KNN
* Apabila cuaca buruk dengan nilai = 1, weekday dan game = 0 maka berarti roti yang harus dibuat ?
* Apabila cuaca baik dengan nilai 4, weekend dan game = 1 maka berarti yang harus dibuat ?

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv("C:/Users/USER/Desktop/PMM-DN-STIMATA MALANG/UAS/dataset_UAS-main/dataset_UAS-main/dataset_soal No.2.csv")
#Kondisi a
query = [1, 0, 0]

#Kondisi b
query1 = [4,1,1]

In [3]:
df.head()

Category   Weather V-1  Holiday V-2  Game V-3  Qty
0         A            5            1         0  250
1         B            3            1         1  200
2         C            1            1         0   75
3         D            4            1         1  400
4         E            4            0         0  150

In [4]:
df.describe()

Weather V-1  Holiday V-2  Game V-3         Qty
count     6.000000     6.000000  6.000000    6.000000
mean      3.166667     0.666667  0.333333  187.500000
std       1.471960     0.516398  0.516398  128.208814
min       1.000000     0.000000  0.000000   50.000000
25%       2.250000     0.250000  0.000000   93.750000
50%       3.500000     1.000000  0.000000  175.000000
75%       4.000000     1.000000  0.750000  237.500000
max       5.000000     1.000000  1.000000  400.000000

In [5]:
#Perhitungan Distance Kondisi a
import math
dis = []
for i in range (6):
    dis.append(math.sqrt((float(df.iloc[i]['Weather V-1']) - (float(query [0]))) **2 + (float(df.iloc[i]['Holiday V-2']) - (float(query[1]))) **2 + (float(df.iloc[i]['Game V-3']) - (float(query[2]))) **2))

In [6]:
df['dis'] = dis
df

Category   Weather V-1  Holiday V-2  Game V-3  Qty       dis
0         A            5            1         0  250  4.123106
1         B            3            1         1  200  2.449490
2         C            1            1         0   75  1.000000
3         D            4            1         1  400  3.316625
4         E            4            0         0  150  3.000000
5         F            2            0         0   50  1.000000

In [7]:
#Yang terdekat ada di kategori B, C, E, F
#Hitung mean
import statistics

produksi = [200, 75, 150, 50]
jumlah = statistics.mean(produksi)
print(jumlah)

118.75


## Result a
Jika cuaca buruk bernilai = 1, weekday bernilai = 0 dan game bernilai = 0 maka jumlah roti yang harus dibuat sebanyak 118.75 atau 119

In [8]:
#Perhitungan Distance Kondisi b
import math
dis1 = []
for i in range (6):
    dis1.append(math.sqrt((float(df.iloc[i]['Weather V-1']) - (float(query1 [0]))) **2 + (float(df.iloc[i]['Holiday V-2']) - (float(query1[1]))) **2 + (float(df.iloc[i]['Game V-3']) - (float(query1[2]))) **2))

In [9]:
df['dis1'] = dis1
df

Category   Weather V-1  Holiday V-2  Game V-3  Qty       dis      dis1
0         A            5            1         0  250  4.123106  1.414214
1         B            3            1         1  200  2.449490  1.000000
2         C            1            1         0   75  1.000000  3.162278
3         D            4            1         1  400  3.316625  0.000000
4         E            4            0         0  150  3.000000  1.414214
5         F            2            0         0   50  1.000000  2.449490

In [10]:
#Yang terdekat ada di kategori A, B, E, F
#Hitung mean
import statistics

produksi1 = [200, 250, 150, 50]
jumlah1 = statistics.mean(produksi1)
print(jumlah1)

162.5


## Result b
Jika cuaca baik bernilai = 4, weekend bernilai = 1 dan game bernilai = 1 maka jumlah roti yang harus dibuat sebanyak 162.5 atau 163